eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxow.hs

```diff
@@ -102,15 +102,17 @@ validateFailedBabbageScripts ::
   ) =>
   Core.Tx era ->
   UTxO era ->
+  Set (ScriptHash (Crypto era)) ->
   Test (Shelley.UtxowPredicateFailure era)
-validateFailedBabbageScripts tx utxo =
+validateFailedBabbageScripts tx utxo neededHashes =
   let failedScripts =
         Map.filterWithKey
           ( \hs script ->
-              let one = isNativeScript @era script
+              let zero = hs `elem` neededHashes
```

Comment:
Not sure if it would be better to compute the intersection of the `neededHashes` with the keys instead of this check

Reply:
This is fine, no need to use `intrersection`, however, I'd suggest switching to `Set.member`, instead of `elem`. This way we'll get a compiler error in an unlikely situation where `neededHashes` switched a type from `Set` to something less efficient, eg. list or Sequence, or even worse a `Map (ScriptHash (Crypto era)) (ScriptHash (Crypto era))`, which would silently change semantics of this function.